crud/common/sharding.lua

```diff
@@ -20,7 +27,22 @@ function sharding.tuple_get_bucket_id(tuple, space, specified_bucket_id)
         return specified_bucket_id
     end
 
-    local key = utils.extract_key(tuple, space.index[0].parts)
+    local primary_index = space.index[0]
```

Comment:
Seems here we should check if ddl available and don't call for each key.

Ideally, we could use code generation here since it requires to choose special subset of key fields (but this part about codegen is optional)

Reply:
```

Reply:
> extract_sharding_key() calculated only when get_ddl_sharding_key() is not nil and additionally latter uses cache.

It doesn't matter because here you use at least next function that can't be jitted and do it always even cache is not empty.